File: backend/app/routes/advisor.py
from fastapi import APIRouter
from datetime import date
from typing import List
from app.services.advisor import Tx, Meta, analizar_usuario

router = APIRouter(prefix="/advisor", tags=["advisor"])

@router.post("/recomendar")
def recomendar(payload: dict):
    # payload esperado: { "txs": [...], "metas": [...] }
    txs_in: List[Tx] = [
        Tx(
            fecha=date.fromisoformat(t["fecha"]),
            monto=abs(float(t["monto"])),
            tipo=t["tipo"],
            categoria=t.get("categoria","Otros"),
            recurrente=bool(t.get("recurrente", False))
        )
        for t in payload.get("txs", [])
    ]
    metas_in: List[Meta] = [
        Meta(
            id=int(m["id"]),
            nombre=m["nombre"],
            monto_objetivo=float(m["monto_objetivo"]),
            ahorro_actual=float(m.get("ahorro_actual",0)),
            fecha_limite=date.fromisoformat(m["fecha_limite"]),
            prioridad_usuario=int(m.get("prioridad_usuario",3))
        )
        for m in payload.get("metas", [])
    ]
    return analizar_usuario(txs_in, metas_in)

File: frontend/lib/advisor.ts
export interface TxPayload {
  fecha: string;
  monto: number;
  tipo: 'ingreso' | 'egreso';
  categoria: string;
  recurrente?: boolean;
}

export interface MetaPayload {
  id: number;
  nombre: string;
  monto_objetivo: number;
  ahorro_actual?: number;
  fecha_limite: string;
  prioridad_usuario?: number;
}

export async function getAdvisorRecommendation(
  fetchWithAuth: (input: RequestInfo, init?: RequestInit) => Promise<Response>,
  txs: TxPayload[],
  metas: MetaPayload[]
) {
  const res = await fetchWithAuth(`${process.env.NEXT_PUBLIC_API_URL}/advisor/recomendar`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ txs, metas }),
  });

  if (!res.ok) {
    throw new Error('Error obteniendo recomendación');
  }
  return res.json();
}
